src/Compilers/CSharp/Portable/Symbols/AssemblySymbol.cs

```diff
@@ -39,6 +40,8 @@ internal abstract class AssemblySymbol : Symbol, IAssemblySymbolInternal
         /// </summary>
         private AssemblySymbol _corLibrary;
 
+        private TypeConversions _lazyTypeConversions;
```

Comment:
>private TypeConversions _lazyTypeConversions; [](http://example.com/codeflow?start=8&length=45)

I think this storage should be moved to ```MetadataOrSourceAssemblySymbol``` and to ```MissingCorLibrarySymbol```. See how special types are cached in ```_lazySpecialTypes```.

Reply:
@AlekseyTs I followed exactly what `_lazySpecialTypes` is doing, but I'm not very confident that the overrides where I have `throw ExceptionUtilities.Unreachable()` are indeed not reachable. Should I instead use `CorLibrary.TypeConversions` instead of those `throw ExceptionUtilities.Unreachable()`?